providers/notificationProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as Toast from '@radix-ui/react-toast';
+import type {
+  Dispatch,
+  FC,
+  PropsWithChildren,
+  ReactNode,
+  SetStateAction,
+} from 'react';
+import { createContext, useEffect, useState } from 'react';
+
+import Notification from '@/components/Common/Notification';
+
+type NotificationContextType = {
+  message: string | ReactNode;
+  duration: number;
+} | null;
+
+type NotificationProps = {
+  className?: string;
+};
+
+const NotificationContext = createContext<NotificationContextType>(null);
+
+export const NotificationDispatch = createContext<
+  Dispatch<SetStateAction<NotificationContextType>>
+>(() => {});
+
+export const NotificationProvider: FC<PropsWithChildren<NotificationProps>> = ({
+  className,
+  children,
+}) => {
+  const [notification, dispatch] = useState<NotificationContextType>(null);
+
+  useEffect(() => {
+    const timeout = setTimeout(() => dispatch(null), notification?.duration);
+
+    return () => clearTimeout(timeout);
+  }, [notification]);
+
+  return (
+    <NotificationContext.Provider value={notification}>
+      <NotificationDispatch.Provider value={dispatch}>
+        <Toast.Provider>
+          {children}
+
+          <Toast.Viewport className="fixed bottom-0 right-0" />
```

Comment:
What I was referring is, can you make this be a `className` call? The reasoning behind is:

- By default those are the classes, but we have an optional prop for the `NotificationProvider` called `viewPortClassName`, this allows us to override the viewport classNames
- This is fundamentally necessary for Storybook, so we can change the NotificationProvider viewPort for Storybook, as for Storybook we want to make the notifications render on the top/left of Storybook instead the usual right/bottom...
- Otherwise we cannot actually assert these notification styles within Chromatic (our Visual Regression Test Provider)

Reply:
And this is what I was referring with the className prop. I don't think we should add a `className` prop on NotificationProvider to style a Notification. (Actually we shouldn't at all do that with the Notifcations that are rendered via the `dispatch` method.